lib/core/application/event/event_application_form_bloc/event_application_form_bloc.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:app/core/domain/event/entities/event.dart';
+import 'package:app/core/domain/user/entities/user.dart';
+import 'package:app/core/utils/string_utils.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'event_application_form_bloc.freezed.dart';
+
+class EventApplicationFormBloc
+    extends Bloc<EventApplicationFormBlocEvent, EventApplicationFormBlocState> {
+  EventApplicationFormBloc()
+      : super(
+          EventApplicationFormBlocState(
+            fieldsState: {},
+            isValid: false,
+          ),
+        ) {
+    on<EventApplicationFormBlocEventInitFieldState>(initFieldState);
+    on<EventApplicationFormBlocEventUpdateField>(updateField);
+  }
+
+  void initFieldState(
+    EventApplicationFormBlocEventInitFieldState event,
+    Emitter emit,
+  ) {
+    final user = event.user;
+    final eventProfileFields = event.event?.requiredProfileFields ?? [];
+
+    final Map<String, String> initialFieldState = {};
+    final userMap = user?.toJson();
+    for (var key in eventProfileFields) {
+      initialFieldState[key] =
+          userMap![StringUtils.snakeToCamel(key)].toString();
+    }
+    emit(state.copyWith(fieldsState: initialFieldState));
+  }
+
+  void updateField(
+    EventApplicationFormBlocEventUpdateField event,
+    Emitter emit,
+  ) {
+    if (event.key == null || event.value == null) {
```

Comment:
can remove this check by define required fields in `EventApplicationFormBlocEventUpdateField`